File: backend/src/main/java/com/musclemap/catalog/MediaSource.java
package com.musclemap.catalog;

/**
 * Where a piece of media is hosted / how it must be rendered. {@code FILE} is a
 * directly addressable URL; {@code YOUTUBE} is a video id embedded via iframe.
 * Mirrors the frontend {@code MediaSource}.
 */
public enum MediaSource {
    FILE,
    YOUTUBE
}

File: frontend/src/domain/enums/MediaSource.ts
/**
 * Where a piece of media is hosted / how it must be rendered.
 * `File` is a directly addressable URL (img/video tag); `YouTube` is a video
 * id embedded via an iframe. New providers (Vimeo, coach uploads…) extend here.
 */
export enum MediaSource {
  File = 'FILE',
  YouTube = 'YOUTUBE',
}
